### ISSUE TITLE
Using an extension overwrites configured delays

### ISSUE THREAD

**Issue by Eamerial** [tomakehurst/wiremock#467]
While the new delay distributions work well, when using an extension the delays do not take effect.

**Comment by tomakehurst** [comment]
Please could you post some details of your setup (code ideally) so that I can try and replicate?

**Comment by Eamerial** [comment]
Hi Tom, Thank You for your quick reply, I really appreciate it.

Unfortunately I'm in a situation where I can't do a remote debug to our running instance of wiremock due to network firewall restrictions :(.

I'm running it as a standalone server with no-request-journal in a docker container on a company server.

Let's do the test with a simple extension:

```
public class ReturnRequest extends ResponseDefinitionTransformer {

    @Override
    public String getName() {
        return "return-request";
    }

    @Override
    public boolean applyGlobally() {
        return false;
    }

    @Override
    public ResponseDefinition transform(Request request, ResponseDefinition responseDefinition, FileSource files, Parameters parameters) {
        return ResponseDefinitionBuilder.like(responseDefinition)
                .but()
                .withBody(request.getBody())
                .withHeader(request.contentTypeHeader().key(), request.contentTypeHeader().firstValue())
                .build();
    }

}
```

Basically a transformer that returns the request back.

Now let's create a simple stub with a uniform distribution delay between 1000 and 5000 ms (so we can really notice the delay) using the REST api:

```
{
    "request" : {
      "url" : "/test/",
      "method" : "POST"
    },
    "response" : {
      "status" : 200,
      "body" : "<?xml version=\"1.0\" encoding=\"UTF-8\"?><test>Hello</test>",
      "headers" : {
        "Content-Type" : "text/xml"
      },
      "delayDistribution":{
        "type" : "uniform",
        "lower" : 1000,
        "upper" : 5000
      }
    }
}
```

if we call wiremock (let's assume it's running at the localhost in port 8080), http://localhost:8080/test/ 
with

```
<?xml version="1.0" encoding="UTF-8" ?>
<root>
<Test>YAY</Test>
</root>
```

we get

```
<?xml version="1.0" encoding="UTF-8" ?>
<test>Hello</test>
```

And sure enough it took about 3 or 4 seconds to reply (different calls and get different delays, as expected).

Now let's create the same stub with our extension as well as the uniform delay between 1000 and 5000 ms :

```
{
    "request" : {
      "url" : "/test/",
      "method" : "POST"
    },
    "response" : {
      "status" : 200,
      "body" : "<?xml version=\"1.0\" encoding=\"UTF-8\"?><test>Hello</test>",
      "headers" : {
        "Content-Type" : "text/xml"
      },
      "delayDistribution":{
        "type" : "uniform",
        "lower" : 1000,
        "upper" : 5000
      },
      "transformers": ["return-request"]
    }
}
```

Now let's call POST http://localhost:8080/test/ again with

```
<?xml version="1.0" encoding="UTF-8" ?>
<root>
<Test>YAY</Test>
</root>
```

and we get

```
<?xml version="1.0" encoding="UTF-8" ?>
<root>
<Test>YAY</Test>
</root>
```

But this time with no delay whatsoever.

Hope it happens the same with you :)

cheers,
v.

**Comment by tomakehurst** [comment]
Thanks for the details. I'll take a look at this.

**Comment by tomakehurst** [comment]
Fixed in 83c2c01b6beafb2d1429af236518adca2dffb45b

**Comment by Eamerial** [comment]
Thanks :)
